clients.go

```diff
@@ -356,6 +357,29 @@ func (cl *Client) Read(packetHandler ReadFn) error {
 	}
 }
 
+// openShutdownSignal initializes a shutdown signal channel for the client.
+// This channel is used to wait for a shutdown notification.
+func (cl *Client) openShutdownSignal() {
+	cl.Lock()
+	defer cl.Unlock()
+	cl.State.shutdownSignal = make(chan any)
+}
```

Comment:
Initialize shutdownSignal as part of NewClient?
Change name to shutdownComplete?

Reply:
NewClient() is called not only when creating new connections but also when called in server.loadClients() to load clients from the database. Clients loaded from the database won't have WriteLoop() and Read() loop. This is why I placed openShutdownSignal() inside attachClient() rather than in NewClient().